### PR TITLE
chore: FollowProfileButton uses cohesion analytics

### PR DESCRIPTION
**Pull request by eessex**
Follow up to #6213, Update `FollowProfileButton` to use cohesion analytics and remove unnecessary props.

### CODE REVIEW

**Comment by eessex** [comment]
bumping for review @damassi @emmadickson

File: src/v2/Apps/Artwork/Components/ArtworkDetails/__tests__/ArtworkDetails.jest.tsx [hunk 1]
```diff
@@ -217,7 +217,7 @@ describe("ArtworkDetails", () => {
           type: "NOT Auction House",
         },
       })
-      expect(wrapper.html()).toContain("Following")
+      expect(wrapper.html()).toContain("Follow")
```
Comment (joeyAghion) [inline src/v2/Apps/Artwork/Components/ArtworkDetails/__tests__/ArtworkDetails.jest.tsx:220]: Are these changes meant to match different button states, or is the label actually "Follow?"

**Review by joeyAghion — APPROVED** [review]

File: src/v2/Apps/Artwork/Components/ArtworkDetails/__tests__/ArtworkDetails.jest.tsx (hunk 1 above)
Comment (joeyAghion) [reply src/v2/Apps/Artwork/Components/ArtworkDetails/__tests__/ArtworkDetails.jest.tsx:220]: I just looked at the artwork page in an attempt to answer my own question and saw multiple "Follow" buttons. If it's important to this test, might be worth scoping the assertion.

Comment (eessex) [reply src/v2/Apps/Artwork/Components/ArtworkDetails/__tests__/ArtworkDetails.jest.tsx:220]: I updated the test fixture to be unfollowed by default, so this changed a few tests that look for text matching. Shouldn't affect any functionality.

**Pull request merged**